src/pages/student/quizzes/QuizList/components/QuestionGrid/index.js

```diff
@@ -1,34 +1,66 @@
-import React from 'react';
+import React, { useState, useEffect } from 'react';
 import { Card } from 'react-bootstrap';
 import { PropTypes } from 'prop-types';
 import { BsClockHistory } from 'react-icons/bs';
 
 import style from './index.module.css';
+import QuizzesTakenReviewApi from '../../../../../../api/QuizTakenReview';
+import QuestionApi from '../../../../../../api/Question';
 
 const QuestionGrid = ({ quiz }) => {
-  const handleCardClick = () => {
-    console.log(`${quiz?.title} is clicked`);
-  };
+  const [QuizzesRecentReview, setQuizzesRecentReview] = useState(null);
+  const [questions, setQuestions] = useState(null);
+
+  useEffect(() => {
+    QuizzesTakenReviewApi.getAll(quiz.id).then(({ data }) => {
+      setQuizzesRecentReview(data.recentQuizzesTaken);
+    });
+
+    QuestionApi.getAll(quiz.id).then(({ data }) => {
+      setQuestions(data.data);
+    });
+  }, []);
+
+  const getHighestScore = () => {
+    if (QuizzesRecentReview?.length > 0) {
+      const highestScore = QuizzesRecentReview.reduce((prev, current) => {
+        return prev.score > current.score ? prev : current;
+      });
 
-  const passOrFailStyle = () => {
-    if (quiz.answerCount > 0 && quiz.correctAnswers > 7) {
-      return style.pass;
+      return highestScore.score;
     }
 
-    if (quiz.answerCount > 0 && quiz.correctAnswers <= 7) {
-      return style.fail;
+    return 0;
+  };
+
+  const getLatestScore = () => {
+    let score = -1;
+    if (QuizzesRecentReview?.length > 0) {
+      QuizzesRecentReview?.forEach((QuizRecent) => {
+        if (quiz.id === QuizRecent.quiz_id && score === -1) {
+          score = QuizRecent.score;
+        }
+      });
+
+      return score;
     }
+  };
 
-    return;
+  const getTotalTimeLimit = () => {
+    if (questions != null) {
+      let total = 0;
+
+      for (let x = 0; x < questions.length; x++) {
+        total += questions[x].time_limit;
+      }
+
+      return total;
+    }
```

Comment:
try to look into es6 "reduce" function for arrays / array of objects and see if you can apply it here.